### PR TITLE
Refactor terraform pieces for deploying osm-stats components to Azure

### PR DESCRIPTION
**Pull request by eternaltyro**
- Refactor terraform code to be compliant with v1.0.11
- Bump AzureRM and random provider versions
- Add lock files to gitignore
- Split the main terraform file into variable, provider and output files
- Fix database resource blocks to update deprecated code
- Use jsonencode for JSON components for better control and readability
- Use hotosm images instead of ARC images
- Use custom OVERPASS_URL value for envvars
- Update formatting and syntax throughout

TODO:

- Upgrade state file using terraform 0.13upgrade

### CODE REVIEW

File: deployment/osm-stats.tf [hunk 1]
```diff
@@ -267,110 +202,102 @@ DEPLOY
...
+      ]
     }
-  ]
-}
-DEPLOY
+  )
 
   parameters = {
-    name = "${var.api_name}"
-    image = "quay.io/americanredcross/osm-stats-api:v0.22.0"
```
Comment (dakotabenjamin) [inline deployment/osm-stats.tf:360]: This label should be kept

File: deployment/osm-stats.tf [hunk 2]
```diff
@@ -1,86 +1,27 @@
...
+  location            = azurerm_resource_group.osm-stats.location
+  resource_group_name = azurerm_resource_group.osm-stats.name
+  ip_address_type     = "Public"
+  os_type             = "Linux"
+  depends_on          = [azurerm_postgresql_database.osm-stats, azurerm_postgresql_firewall_rule.osm-stats]
 
   container {
-    name = "osm-changes"
-    image = "docker.io/hotosm/osm-stats-workers:inc-timeout"
-    cpu = "3"
```
Comment (dakotabenjamin) [inline deployment/osm-stats.tf:83]: any particular reason to lower this value?

**Review by dakotabenjamin — CHANGES_REQUESTED** [review]
I was able to run a terraform plan using v0.13. There are a few changes needed, but then let's discuss next steps.

**Comment by eternaltyro** [comment]
@dakotabenjamin made the reverts

**Comment by eternaltyro** [comment]
@dakotabenjamin fixed a couple more lines for syntactic sugar.

**Pull request closed without merging**